trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/library/MathLibraryModel.java

```diff
@@ -532,6 +596,464 @@ private Result handleIsLessGreater(final IDispatcher main, final IASTFunctionCal
+				arguments.get(0).getLrValue().getValue(), arguments.get(1).getLrValue().getValue(), type), type))
+				.build();
+	}
+
+	private ExpressionResult handleFdim(final IDispatcher main, final IASTFunctionCallExpression node,
+			final ILocation loc, final String name, final CPrimitive type) {
+		final List<ExpressionResult> arguments = handleFloatArguments(main, node, loc, name, 2, type);
+		final Expression first = arguments.get(0).getLrValue().getValue();
+		final Expression second = arguments.get(1).getLrValue().getValue();
+		final Expression comparison = mExpressionTranslation.constructBinaryComparisonExpression(loc,
+				IASTBinaryExpression.op_greaterThan, first, type, second, type);
+		final Expression subtraction = mExpressionTranslation.constructArithmeticExpression(loc,
+				IASTBinaryExpression.op_minus, first, type, second, type);
+		final Expression zero = mExpressionTranslation.constructLiteralForFloatingType(loc, type, BigDecimal.ZERO);
+		final Expression resultExprFdim =
+				ExpressionFactory.constructIfThenElseExpression(loc, comparison, subtraction, zero);
+		final Expression secondNaNExpr = ExpressionFactory.constructIfThenElseExpression(loc,
+				mFloatHandler.isNan(loc, second, type), second, resultExprFdim);
+		final Expression firstNaNExpr = ExpressionFactory.constructIfThenElseExpression(loc,
+				mFloatHandler.isNan(loc, first, type), first, secondNaNExpr);
+		return new ExpressionResultBuilder().addAllExceptLrValue(arguments).setLrValue(new RValue(firstNaNExpr, type))
+				.build();
+	}
+
+	private ExpressionResult handleSignbit(final IDispatcher main, final IASTFunctionCallExpression node,
+			final ILocation loc, final String name) {
+		final IASTInitializerClause[] arguments = node.getArguments();
+		mHelper.checkArguments(loc, 1, name, arguments);
+		final ExpressionResult argumentResult =
+				mExprResultTransformer.transformDispatchDecaySwitchRexBoolToInt(main, loc, arguments[0]);
+		final CPrimitive intType = new CPrimitive(CPrimitives.INT);
+		final Expression argument = argumentResult.getLrValue().getValue();
+		final CPrimitive type = (CPrimitive) argumentResult.getCType();
+		final ExpressionResultBuilder builder = new ExpressionResultBuilder().addAllExceptLrValue(argumentResult);
+		final AuxVarInfo auxVar = mAuxVarInfoBuilder.constructAuxVarInfo(loc, intType, AUXVAR.RETURNED);
+		builder.addAuxVarWithDeclaration(auxVar);
+		final Statement nondet = new HavocStatement(loc, new VariableLHS[] { auxVar.getLhs() });
+		// TODO: Handle negative NaN correctly, only overapproximated until then
+		// signbit(x) := isNegative(x) ? 1 : 0;
+		new OverapproxVariable("sign of NaN", loc).annotate(nondet);
+		final Expression zero = mExpressionTranslation.constructLiteralForIntegerType(loc, intType, BigInteger.ZERO);
+		builder.addStatement(StatementFactory.constructIfStatement(loc, mFloatHandler.isNan(loc, argument, type),
```

Comment:
@danieldietsch @Heizmann As you have worked with floats in the past, does this handling of `signbit` (incl. the overapproximation) seem reasonable?